### PR TITLE
DOECLIM integration

### LINKED ISSUE #505: Ocean CMIP temperature data needed
We need some data for Hector's ocean component, these values will most likely come from a multi-model CMIP mean. 

* pre-industrial weighted mean ocean surface temperature (for total surface) 
* relationship between total mean ocean surface temperature & the following 
     * ocean surface temp for HL (defined as latitude > 55)
     * ocean surface temp for LL (defined as latitude < 55)

### PR DESCRIPTION
**Pull request by kdorheim**
Should close #505, #444, #68, #304, #69, it is based off on #389. 

**Type of development** 

MAJOR: new capabilities & changes in Hector behavior 

* Couple DOECLIM output with other Hector components. Simple N box uses land air temp. Ocean component uses SST. 
* Implement user-defined land-ocean warming ratio. 
* Add total ocean outputs for the surface ocean results
* General clean up (remove variables that aren't being used)

**Checklist**
- [x] [Roxygen](https://roxygen2.r-lib.org/) where applicable but also thorough and coherent inline documentation 
- [x] [Unit tests](https://r-pkgs.org/tests.html) when applicable
- [x] Complies with the [Hector code style guide](https://jgcri.github.io/hector/articles/manual/StyleGuide.html)
- [x] The PR passes automated testing suite

**Problems** 
 Right now this PR fails _many_ unit tests. I suspect that some of this stems from using Tgav land in simple N box instead of Tgav. PR needs to be able to pass tests before merging. @bpbond :( 

# Changes in Hector output

## DOECLIM 

Full DOECLIM integration analysis can be found [here](https://rpubs.com/kdorheim/863638).

image

image

image

## LO Warming Ratio

More details can be found [here](https://rpubs.com/kdorheim/864447)

image

## New Outputs 

image

image

image

### CODE REVIEW

File: R/RcppExports.R [hunk 1]
```diff
@@ -1163,6 +1169,12 @@ PH_LL <- function() {
     .Call('_hector_PH_LL', PACKAGE = 'hector')
 }
 
+#' @describeIn ocean Ocean surface Ph
```
Comment (bpbond) [inline R/RcppExports.R:1172]: "pH"

File: R/RcppExports.R [hunk 2]
```diff
@@ -1187,25 +1199,37 @@ PCO2_LL <- function() {
...
+}
+
 #' @describeIn ocean Dissolved inorganic carbon, low-latitude
 #' @export
 DIC_LL <- function() {
     .Call('_hector_DIC_LL', PACKAGE = 'hector')
 }
 
-#' @describeIn ocean Ocean temperature, high-latitude
+#' @describeIn ocean Absolute ocean surface temperature, high-latitude (deg C)
```
Comment (bpbond) [inline R/RcppExports.R:1226]: Following up on our conversation the other day, is this (absolute) correct here?

File: R/RcppExports.R [hunk 3]
```diff
@@ -1429,6 +1459,14 @@ LAND_AIR_TEMP <- function() {
     .Call('_hector_LAND_AIR_TEMP', PACKAGE = 'hector')
 }
 
+#' @describeIn parameters Land-Ocean Warming Ratio (\code{"(unitless)"}), by default set to 9999 (NULL)
+#' meaning that the land ocean warming ratio is an emergent property of Hector's temperature
+#' component otherwise the user defined land ocean warming ratio will be used to determine
```
Comment (bpbond) [inline R/RcppExports.R:1464]: Maybe just "...will be used"

File: inst/include/oceanbox.hpp [hunk 4]
```diff
@@ -27,9 +27,10 @@
 #include "ocean_csys.hpp"
 #include "fluxpool.hpp"
 
-// Mean absolute global temperature, preindustrial (deg C), this is used by the ocean component which, requires
-// absolute temperature to calculate ocean chemistry.
-#define MEAN_GLOBAL_TEMP 15
+// Mean absolute global tos temperature, preindustrial (deg C), this is used by the ocean component which, requires
+// absolute temperature to calculate ocean chemistry, the CMIP6 multi model mean from 1850 - 1860. See hector_cmip6data
+// for details. TODO tag a zenondo archive prior to Hector v 3 release.
```
Comment (bpbond) [inline inst/include/oceanbox.hpp:32]: Maybe move this TODO to an issue

File: inst/include/temperature_component.hpp [hunk 5]
```diff
@@ -145,6 +145,18 @@ class TemperatureComponent : public IModelComponent {
     unitval flux_interior; //!< heat flux into interior layer of ocean, W/m2
     unitval heatflux;      //!< heat flux into ocean, W/m2
 
+    // Potenital user defined model parameters
```
Comment (bpbond) [inline inst/include/temperature_component.hpp:148]: "Potential"

File: src/temperature_component.cpp [hunk 6]
```diff
@@ -207,6 +217,11 @@ void TemperatureComponent::prepareToRun() {
         H_LOG( glog, Logger::WARNING ) << "Temperature will be overwritten by user-supplied values!" << std::endl;
     }
 
+    if( lo_warming_ratio != 9999.0) {
```
Comment (bpbond) [inline src/temperature_component.cpp:220]: Ugh, I really don't like this kind of magic value

File: inst/include/temperature_component.hpp [hunk 7]
```diff
@@ -145,6 +145,18 @@ class TemperatureComponent : public IModelComponent {
...
+    unitval lo_warming_ratio; //!< user defined land ocean warming ratio, unitless
```
Comment (bpbond) [inline inst/include/temperature_component.hpp:149]: Let's provide some more information here? E.g. 
```cpp
// This ratio is the defined as ΔTL/ΔTO; for more information see
// https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/grl.50971
```

Comment (bpbond) [reply inst/include/temperature_component.hpp:149]: ☝️  and/or in the ini file?

File: R/RcppExports.R (hunk 2 above)
Comment (kdorheim) [reply R/RcppExports.R:1226]: Yup!

File: inst/include/temperature_component.hpp (hunk 7 above)
Comment (kdorheim) [reply inst/include/temperature_component.hpp:149]: Can totally add that comment to the code. Would we want to switch from unitval U_UNITLESS to U_DTL_DTO or something? Or is it okay to leave at unitval unitless?

Comment (bpbond) [reply inst/include/temperature_component.hpp:149]: I think unitless is correct.

**Comment by kdorheim** [comment]
@bpbond have the requested changes been resolved?

**Review by bpbond — APPROVED** [review]

**Pull request merged**